js_modules/dagster-ui/packages/ui-core/src/performance/TraceContext.tsx

```diff
@@ -59,11 +59,11 @@ type Options = {
 
 /** Use this to declare a dependency on an apollo query result */
 export function useBlockTraceOnQueryResult(
-  queryResult: Pick<QueryResult<any, any>, 'data' | 'error'>,
+  queryResult: Pick<QueryResult<any, any>, 'data' | 'error' | 'called'>,
   name: string,
   opts: Options = {},
 ) {
-  const dep = useTraceDependency(name, opts);
+  const dep = useTraceDependency(name, {skip: opts.skip || !queryResult.called});
```

Comment:
Added `!queryResult.called` here which makes it so that we can do

`useBlockTraceOnQueryResult(lazyQueryResult, 'LazyQuery')`
without actually blocking the trace if the lazy query is never actually called.